### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by olanikegloria**
I'm excited to present the following updates in this pull request:

1. **Authentication Enhancement**
   - Removed the custom `current_user` method from `ApplicationController` as Devise now handles this functionality.

2. **Devise Integration and Configuration**
   - Integrated and configured Devise, allowing for seamless user authentication.

3. **User Management Enhancements**
   - Implemented user registration functionality, enabling new users to register securely.
   - Enabled users to log in using their email and password, with passwords securely hashed and stored in the database.
   - Implemented email confirmation to ensure a valid email for each user.
   - Implemented a password reset feature to enhance user experience.

### CODE REVIEW

File: app/views/devise/confirmations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<h2>Resend confirmation instructions</h2>
+
+<%= form_for(resource, as: resource_name, url: confirmation_path(resource_name), html: { method: :post }) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
```
Comment (TadesseDev) [inline app/views/devise/confirmations/new.html.erb:4]: - [ ] Dear Team, for this project we need to follow the [getting started link](https://github.com/heartcombo/devise) to install and set up the device gem for authentication. I can see your changes only contain files related to the view file. If you follow the setup guide properly you will be able to set up your model and other appropriate setup which are required for setting up the device properly. Please make sure to follow the setup process closely.

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<h2>Log in</h2>
+
+<%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
+  <div class="field">
```
Comment (TadesseDev) [inline app/views/devise/sessions/new.html.erb]: - [ ] Good job adding the view files team, but it's expected the application allows for registering a new user and allowing login for a registered user. Currently, I cannot access the route for the login and sign-up pages. Please make sure these features are in place and we can register and log in as a user.

File: config/initializers/devise.rb [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+# is set to its default value. Note that some are commented out while others
+# are not: uncommented lines are intended to protect your configuration from
+# breaking changes in upgrades (i.e., in the event that future versions of
+# Devise change the default values for those options).
+#
+# Use this hook to configure devise mailer, warden hooks and so forth.
+# Many of these configuration options can be set straight in your model.
+Devise.setup do |config|
+  # The secret key used by Devise. Devise uses this key to generate
+  # random tokens. Changing this key will render invalid all existing
```
Comment (TadesseDev) [inline config/initializers/devise.rb:13]: - [ ] Alongside those features we need to enable some features inside our models/user.rb fille. This features are essential requirements and we need to follow them closely. Please enable the following device modules ```
database_authenticatable, :registerable,:recoverable, :rememberable, :validatable, :confirmable```. we need to make sure `A hashed password is stored in the database.`, `The application asks for confirmation of email.`, `A user can reset the password in the application.`,

File: app/views/devise/registrations/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+<h2>Sign up</h2>
+
+<%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+
```
Comment (TadesseDev) [inline app/views/devise/registrations/new.html.erb]: - [ ] One last thing is, that I could not initialize the app and access the login and signup pages, Once you set the devise Gem properly, Please make sure the views in the devise for the registration and login are modified to match the wireframes and their styling.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] View files are added in the view/device folder ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
- [ ] Dear team, Please keep in mind device provides a method to provide the `current_user` method, hence the `current_user` method in `ApplicationController` is expected to be been removed. 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @olanikegloria and @MNisarAli , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented the changes requested to ensure devise is setup correctly, and the user is able to access the login and signup pages.
✔️ Ensured the login and signup pages follows the same design given in the wireframe

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**